File: AABB/AABB.lua
local M_M = {
	__call = function(class, ...)
		return class.new(...)
	end
}
O_M = {}
O_I = {
	clone = function(self)
		return M({unpack(self.point)}, {unpack(self.extent)})
	end,
	contains = function(self, point)
		for i = 1,#self.point do
			if point[i] < self.point[i] then return false end
			if point[i] > self.point[i] + self.extent[i] then return false end
		end
		return true
	end,
	intersects = function(self, b)
		for i = 1,#self.point do
			if b.point[i] > self.point[i] + self.extent[i] then return false end
			if b.point[i] + b.extent[i] < self.point[i] then return false end
		end
		return true
	end
}
local M = {}
	M.new = function(point, extent)
		if #point ~= ~extent then
			error("AABB: Constructor: parameter size mismatch")
		end

		local self = setmetatable({}, O_M)
		self.point = point
		self.extent = extent
		return self
	end

O_M.__index = O_I
setmetatable(M, M_M)

return M

File: AABB/AABB.py
#!/usr/bin/python3

class AABB:
	def __init__(self, point, extent):
		self.point = point
		self.extent = extent

	@staticmethod
	def contains(aabb, point):
		if len(aabb.point) != len(point): return False
		for i in range(len(point)):
			if (point[i] > aabb.point[i] and
					point[i] < aabb.point[i] + aabb.extent[i]):
				return True
		return False
	@staticmethod
	def intersects(r1, r2):
		for i in range(len(r1)):
			if ((r1.point[i] > r2.point[i] + r2.extent[i]) or
					(r2.point[i] > r1.point[i] + r1.extent[i])):
				return False
		return True
